src/utils/format.test.js

```diff
@@ -0,0 +1,121 @@
+/* global BigInt */
+
+import { formatIntegerRange, formatNumber, formatTokenAmount } from './format'
+import { NO_BREAK_SPACE } from './characters'
+
+describe('formatIntegerRange()', () => {
+  test('should constrain the result in a range', () => {
+    expect(formatIntegerRange(111)).toEqual('99')
+    expect(formatIntegerRange(111, 0, 9)).toEqual('9')
+    expect(formatIntegerRange(-234)).toEqual('0')
+  })
+  test('should add the suffix as needed', () => {
+    expect(formatIntegerRange(9, 0, 9, '+')).toEqual('9')
+    expect(formatIntegerRange(10, 0, 9, '+')).toEqual('9+')
+  })
+})
+
+describe('formatNumber()', () => {
+  test('should add commas', () => {
+    expect(formatNumber(1)).toEqual('1')
+    expect(formatNumber(12)).toEqual('12')
+    expect(formatNumber(123)).toEqual('123')
+    expect(formatNumber(1234)).toEqual('1,234')
+    expect(formatNumber(12345)).toEqual('12,345')
+    expect(formatNumber(123456)).toEqual('123,456')
+    expect(formatNumber(1234567)).toEqual('1,234,567')
+  })
+
+  test('should work with decimals', () => {
+    expect(formatNumber(1.3)).toEqual('1.3')
+    expect(formatNumber(12.12)).toEqual('12.12')
+    expect(formatNumber(123.123)).toEqual('123.123')
+    expect(formatNumber(1234.1234)).toEqual('1,234.1234')
+    expect(formatNumber(12345.12345)).toEqual('12,345.12345')
+    expect(formatNumber(123456.123456)).toEqual('123,456.123456')
+    expect(formatNumber(1234567.1234567)).toEqual('1,234,567.1234567')
+  })
+
+  test('should work with strings', () => {
+    expect(formatNumber('123456123456')).toEqual('123,456,123,456')
+    expect(formatNumber('1234567.1234567')).toEqual('1,234,567.1234567')
+  })
+
+  test('should work with BigInt', () => {
+    expect(formatNumber(BigInt('123456123456239873298739287'))).toEqual(
+      '123,456,123,456,239,873,298,739,287'
+    )
+  })
+})
+
+describe('formatTokenAmount()', () => {
+  test('should handle native numbers', () => {
+    expect(formatTokenAmount(9381295879707883945, 18, 1)).toEqual('9.4')
+  })
+
+  test('should format numbers properly', () => {
+    expect(formatTokenAmount('32989381295879707883945', 18, 4)).toEqual(
+      '32,989.3813'
+    )
+  })
+
+  test('should handle string numbers', () => {
+    expect(formatTokenAmount('2839381295879707883945', 18, 18)).toEqual(
+      '2,839.381295879707883945'
+    )
+  })
+
+  test('should handle BigInt-like numbers', () => {
+    expect(formatTokenAmount(BigInt('2839381295879707883945'), 18, 8)).toEqual(
+      '2,839.38129588'
+    )
+  })
+
+  test('should handle non-decimal numbers', () => {
```

Comment:
```suggestion
  test('should handle large integers', () => {
```

Or maybe "large numbers without decimals"?

Reply:
https://github.com/aragon/aragon-ui/pull/728/commits/878fa619d98824eb37c22db62d3fcc68bcf1ecc1